### ISSUE TITLE
Autonotes and global flags refactoring proposal

### ISSUE THREAD

**Issue by DominusExult** [exult/exult#399]
With the help of documents found in the Origin Museum I reconstructed all of BG global flags and added the ones we already named from FoV (not complete) - attached to this issue. I used mostly the short names from the documentation unless the ones we already used makes more sense. Some make no sense to me and some are unused by Exult.

[bg_gflags.txt](https://github.com/exult/exult/files/12833031/bg_gflags.txt)

Right now we have two places that show the global flags: the /data/bg/autonotes.txt (/data/si/autonotes.txt) and /content/bgkeyring/src/headers/bg/bg_gflags.uc bg_gflags2.uc (/content/sifixes/src/header/si/si_gflags.uc).

Both are not ideal. Autonotes should ideally only list the nicely written description for the notebook and the mods have the list as usecode.

My proposal:
- rename autonotes.txt to bg_gflags.txt and si_gflags.txt
- instead of using the autonotes.txt formatting
`0x3c:What's this? A body! And a key? Could this be a clue?`, write
`0x3C:ENTERED_STABLES {Set when you enter the Trinsic stables the first time} "What's this? A body! And a key? Could this be a clue?"` (<- proposal)
- in the cheat screen global flags editor show the short name `ENTERED_STABLES` (and if the global flag number has no matching entry, display `UNKNOWN`)
- in the notebook record the prose `What's this? A body! And a key? Could this be a clue?` (as it does already, only for those flags that have an entry). For flags without prose, show the short name(?).
- some flags are purely game mechanics, so it might make sense to have something in the entry to prevent them from being recorded in the notebook. Maybe something like `0x1F:WHEEL_1 {no idea what it does} NO_RECORD`

I can format the text file in whatever way we decide but I can't make the changes to the notebook gump or the cheat screen.

There is an ages old discussion about this at http://exult.info/forum/viewtopic.php?f=1&t=136892&p=1715679&hilit=global+flags#p1715679 with @malignantmanor and @KnightCaptainU7

**Comment by wench** [comment]
This sounds like a really good idea for me

**Comment by KnightCaptainU7** [comment]
I like the idea of showing the short name in the cheat menu.

Coding-wise, I think approaching the cheat menu short name list separately
from the autonotes would be easier to do. The short name list won't see as
much change over time.

This may invite writers to submit autonotes by using a simplied autonotes
format.

**Comment by marzojr** [comment]
> I like the idea of showing the short name in the cheat menu. Coding-wise, I think approaching the cheat menu short name list separately from the autonotes would be easier to do. The short name list won't see as much change over time. This may invite writers to submit autonotes by using a simplied autonotes format.

I agree with this. We could also amend UCC to emit a gflags section of the symbol table containing names declared in usecode, or maybe even add the way to specify the whole thing in Usecode.

**Comment by KnightCaptainU7** [comment]
Hi Dominus. I have a massive list of SI flags, and they are named in SI
Fixes friendly names. Short descriptions are helpful.

**Comment by KnightCaptainU7** [comment]
[si_gflags.use.txt](https://github.com/user-attachments/files/25455262/si_gflags.use.txt)

Here's my list which was long-ago reordered numerically. To my knowledge, this is the most complete list. Gflags 2, 41, and any after 792 are not in the DOS release of SI+SS.

**Comment by DominusExult** [comment]
> I wonder if we should add a way to declare flags in UCC that allow tagging them with an autonotes field? Then usecode could have entries for them in the symbol table, and we would just need autonotes for the original games.

That is a good idea.

I've completed this issue and will add a separate one for the UCC autonotes :)

**Comment by KnightCaptainU7** [comment]
> I would love to compare this to the list I have on [this repository](https://github.com/marzojr/u7-usecode) I just made public.

Feel free to cross-pollinate from my list. It looks like we did something similar to the decompiled Usecode. I used a giant SED script, have a lot of the functions identified and named.

[si_replacements.txt](https://github.com/user-attachments/files/25462796/si_replacements.txt)

More functions were identified after that SED script from early 2017, but I didn't SED them up.
